### PR TITLE
Some string fixes

### PR DESCRIPTION
**Pull request by kpoeck**
Return correct indices for the string functions

### CODE REVIEW

File: src/core/array.cc [hunk 1]
```diff
@@ -2057,6 +2059,7 @@ Array_sp SimpleBitVector_O::unsafe_setf_subseq(size_t start, size_t end, Array_s
   TYPE_ERROR(other,cl::_sym_bit_vector);
 }
 
+// This does not work properly
```
Comment (drmeister) [inline src/core/array.cc]: Does it need to be removed or fixed?

**Comment by kpoeck** [comment]
Needs to be fixed, see issue 436

**Comment by kpoeck** [comment]
The rest of the fixes should be fine though

**Comment by kpoeck** [comment]
@drmeister , have now fixed Fix SimpleBitVector_O::unsafe_fillArrayWithElt, so all should be fine now. 
Tested with more regression-tests and ansi-tests

**Comment by drmeister** [comment]
Merged - thank you!

**Pull request merged**